packages/mermaid/src/diagrams/flowchart/flowDb.ts

```diff
@@ -762,6 +816,21 @@ export const lex = {
 };
 
 const getTypeFromVertex = (vertex: FlowVertex) => {
+  if (vertex?.img) {
+    return 'imageSquare';
+  }
+  if (vertex?.icon) {
+    if (vertex.form === 'circle') {
+      return 'iconCircle';
+    }
+    if (vertex.form === 'square') {
+      return 'iconSquare';
+    }
+    if (vertex.form === 'rounded') {
+      return 'iconRounded';
+    }
+    return 'icon';
+  }
   if (vertex.type === 'square') {
```

Comment:
nit: Why do we `?` on vertex on some places only?
We should remove from everywhere, and use an early return if reqired. 

There is no runtime safety now, as even if `vertex?.icon` is safe, `vertex.type` check will fail.

Reply:
I'll update to `vertex?.form`. Early return is not possible here. We need to check if form is circle square or rounded. if there is anything else will return icon.

Reply:
By early return, I meant return early if vertex is undefined. Then we don't need `?` everywhere.

Reply:
Ohh this will not work. say someone added form as `xyx` which is not valid. when we do early return we will not return shape as icon.

Reply:
```ts
  if (vertex?.img) {
  if (vertex?.icon) {
  if (vertex.type === 'square') {
```

I'm not talking about the internals. Consider the above, the first 2 have `?`, while the 3rd one does not. So, there is no actual runtime safety if `vertex` is `undefined`, as even if the first two passes, third one will throw error as we're trying to access `type` of `undefined`. 

So, either we should enforce properly that vertex is never undefined, and remove the `?`s.
Or, we could do `if(!vertex) return`, and then remove the `?`s, or add `?` to all the required usecases.

Reply:
Ohh my bad I taught you are taking about form or icon is undefined. Vertex will be always defined. I can remove `?`